src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/ProvenBlockHeaderStore.cs

```diff
@@ -273,7 +274,17 @@ public async Task SaveAsync()
                 using (new StopwatchDisposable(o => this.performanceCounter.AddInsertTime(o)))
                 {
                     // Save the items to disk.
-                    await this.provenBlockHeaderRepository.PutAsync(pendingBatchInsert, hashHeight).ConfigureAwait(false);
+                    try
+                    {
+                        await this.provenBlockHeaderRepository.PutAsync(pendingBatchInsert, hashHeight).ConfigureAwait(false);
+                    }
+                    catch (TableNotOperableException ex)
```

Comment:
Normally DB retry logic done in the layer that is dedicated to the specific implementation, for example catching a `TableNotOperableException` is specific to DBfreeze this ties the high level component to the DBreeze implementation.

Reply:
@dangershony Moved logic to the underlying repo